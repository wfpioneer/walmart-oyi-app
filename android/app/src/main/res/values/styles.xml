<resources>

    <!-- Base application theme. -->
    <style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">
        <!-- Customize your theme here. -->
        <item name="android:textColor">#000000</item>
        <item name="android:statusBarColor">@color/blue</item>
        <item name="android:datePickerDialogTheme">@style/Dialog.Theme</item>
        <item name="android:timePickerDialogTheme">@style/Dialog.Theme</item>
    </style>

    <style name="SplashTheme" parent="Theme.AppCompat.Light.NoActionBar">
        <item name="android:windowBackground">@drawable/background_splash</item>
        <item name="android:statusBarColor">@color/splashscreen_bg</item>
    </style>

     <!-- Configuration for DatePicker Dialog theme. -->
    <style name="Dialog.Theme" parent="Theme.AppCompat.Light.Dialog">
        <!-- BackgroundColor of the date modal -->
        <item name="colorAccent">@color/blue</item>
        <!-- textColor of the date modal -->
        <item name="android:textColorPrimary">#000000</item>
    </style>

</resources>
