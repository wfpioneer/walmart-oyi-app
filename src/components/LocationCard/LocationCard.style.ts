import { StyleSheet } from 'react-native';
import COLOR from '../../themes/Color';

const center = 'center';
export default StyleSheet.create({
  mainContainer: {
    backgroundColor: COLOR.WHITE
  },
  container: {
    alignItems: center,
    justifyContent: 'space-evenly',
    paddingHorizontal: 10,
    paddingVertical: 8,
    flexDirection: 'row'
  },
  content: {
    flex: 1,
    justifyContent: center,
    alignItems: center
  },
  location: {
    fontSize: 16,
    fontWeight: 'bold'
  },
  mandatoryLocScan: {
    fontSize: 16,
    color: COLOR.RED_400,
    fontWeight: 'bold'
  },
  locScanned: {
    fontSize: 16,
    color: COLOR.GREEN,
    fontWeight: 'bold'
  },
  pallet: {
    fontSize: 12,
    marginTop: 1
  },
  mandatoryPalletScan: {
    fontSize: 12,
    color: COLOR.RED_400,
    marginTop: 1
  },
  palletScanned: {
    fontSize: 12,
    color: COLOR.GREEN,
    marginTop: 1
  },
  actionContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    alignItems: center
  },
  errorText: {
    fontSize: 14,
    color: COLOR.RED_700,
    textAlign: center,
    paddingBottom: 5
  }
});
