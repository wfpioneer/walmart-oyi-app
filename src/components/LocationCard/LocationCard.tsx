import React from 'react';
import {
  Text,
  TouchableOpacity,
  View
} from 'react-native';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import { strings } from '../../locales';
import NumericSelector from '../NumericSelector/NumericSelector';
import styles from './LocationCard.style';
import COLOR from '../../themes/Color';

interface LocationCardProp {
    location: string;
    palletID?: string;
    quantity: number;
    locationType: string;
    onQtyIncrement(): void;
    onQtyDecrement(): void;
    onTextChange(text: string): void;
    onLocationDelete(): void;
    onEndEditing?: () => void;
    scannerEnabled: boolean;
    scanned?: boolean;
  }

export const MAX = 9999;
export const MIN = 0;

export const getContentStyle = (
  locationType: string, scannerEnabled: boolean, scanned: boolean | undefined, isLocationText: boolean
) => {
  if (locationType === 'reserve' && scannerEnabled && scanned) {
    return isLocationText ? styles.locScanned : styles.palletScanned;
  }
  if (locationType === 'reserve' && scannerEnabled && !scanned) {
    return isLocationText ? styles.mandatoryLocScan : styles.mandatoryPalletScan;
  }
  return isLocationText ? styles.location : styles.pallet;
};

export const validateQty = (qty: number) => MIN <= qty && qty <= MAX;

const LocationCard = (props: LocationCardProp): JSX.Element => {
  const {
    location,
    palletID,
    quantity,
    locationType,
    onQtyIncrement,
    onQtyDecrement,
    onTextChange,
    onLocationDelete,
    onEndEditing,
    scannerEnabled,
    scanned
  } = props;
  return (
    <View style={styles.container}>
      <View style={styles.content}>
        <View>
          <Text style={getContentStyle(locationType, scannerEnabled, scanned, true)}>
            {location}
          </Text>
        </View>
        {locationType === 'reserve' && (
        <View>
          <Text style={getContentStyle(locationType, scannerEnabled, scanned, false)}>
            {`${strings('LOCATION.PALLET')} ${palletID}`}
          </Text>
        </View>
        )}
      </View>
      <View style={styles.actionContainer}>
        <View>
          <NumericSelector
            onDecreaseQty={onQtyDecrement}
            onIncreaseQty={onQtyIncrement}
            onTextChange={onTextChange}
            minValue={0}
            maxValue={MAX}
            value={quantity}
            isValid={validateQty(quantity)}
            onEndEditing={onEndEditing}
          />
        </View>
        <View>
          <TouchableOpacity onPress={() => { onLocationDelete(); }}>
            <MaterialCommunityIcons name="trash-can" size={40} color={COLOR.TRACKER_GREY} />
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
};

LocationCard.defaultProps = {
  palletID: '',
  scanned: false,
  onEndEditing: () => {}
};

export default LocationCard;
