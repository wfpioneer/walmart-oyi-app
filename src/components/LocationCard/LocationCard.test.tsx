import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import LocationCard, { getContentStyle } from './LocationCard';
import styles from './LocationCard.style';

describe('LocationCard', () => {
  describe('Tests rendering LocationCard', () => {
    const renderer = ShallowRenderer.createRenderer();
    it('LocationCard with valid input', () => {
      renderer.render(<LocationCard
        location="A1-1"
        locationType="floor"
        onQtyIncrement={jest.fn}
        onEndEditing={jest.fn}
        onTextChange={jest.fn}
        onQtyDecrement={jest.fn}
        palletID="1234"
        scannerEnabled={false}
        scanned={false}
        quantity={12}
        onLocationDelete={jest.fn}
      />);
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });

    it('LocationCard with locationType as reserve', () => {
      renderer.render(<LocationCard
        location="A1-1"
        locationType="reserve"
        onQtyIncrement={jest.fn}
        onEndEditing={jest.fn}
        onTextChange={jest.fn}
        onQtyDecrement={jest.fn}
        palletID="1234"
        scannerEnabled={false}
        scanned={false}
        quantity={12}
        onLocationDelete={jest.fn}
      />);
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });

    it('renders the card with locationType reserve and scannerEnabled true', () => {
      renderer.render(<LocationCard
        location="A1-1"
        locationType="reserve"
        onQtyIncrement={jest.fn}
        onEndEditing={jest.fn}
        onTextChange={jest.fn}
        onQtyDecrement={jest.fn}
        palletID="1234"
        scannerEnabled={true}
        scanned={false}
        quantity={12}
        onLocationDelete={jest.fn}
      />);
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });

    it('renders the card with locationType reserve, scannerEnabled true and scanned true', () => {
      renderer.render(<LocationCard
        location="A1-1"
        locationType="reserve"
        onQtyIncrement={jest.fn}
        onEndEditing={jest.fn}
        onTextChange={jest.fn}
        onQtyDecrement={jest.fn}
        palletID="1234"
        scannerEnabled={true}
        scanned={true}
        quantity={12}
        onLocationDelete={jest.fn}
      />);
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
  });

  describe('Tests getContentStyle function', () => {
    it('should return style location when location type floor and isLocation content true', () => {
      expect(getContentStyle('Floor', false, false, true)).toStrictEqual(styles.location);
    });
    it('should return style pallet when location type floor and isLocation content false', () => {
      expect(getContentStyle('Floor', false, false, false)).toStrictEqual(styles.pallet);
    });
    it('should return style mandatorylocation when location type reserve, isLocation content true', () => {
      expect(getContentStyle('reserve', true, false, true)).toStrictEqual(styles.mandatoryLocScan);
    });
    it('should return style mandatoryPallet when location type reserve, isLocation content false', () => {
      expect(getContentStyle('reserve', true, false, false)).toStrictEqual(styles.mandatoryPalletScan);
    });
    it('should return style locationscanned when location type reserve, isLocation content true', () => {
      expect(getContentStyle('reserve', true, true, true)).toStrictEqual(styles.locScanned);
    });
    it('should return style mandatoryPallet when location type reserve, isLocation content false', () => {
      expect(getContentStyle('reserve', true, true, false)).toStrictEqual(styles.palletScanned);
    });
  });
});
