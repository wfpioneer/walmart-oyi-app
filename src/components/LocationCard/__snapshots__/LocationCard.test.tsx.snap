// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationCard Tests rendering LocationCard LocationCard with locationType as reserve 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffffff",
      "flexDirection": "row",
      "justifyContent": "space-evenly",
      "paddingHorizontal": 10,
      "paddingVertical": 8,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        A1-1
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "fontSize": 12,
            "marginTop": 1,
          }
        }
      >
        [missing "en.LOCATION.PALLET" translation] 1234
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
  >
    <View>
      <NumericSelector
        isValid={true}
        maxValue={9999}
        minValue={0}
        onDecreaseQty={[Function]}
        onEndEditing={[Function]}
        onIncreaseQty={[Function]}
        onTextChange={[Function]}
        value={12}
      />
    </View>
    <View>
      <TouchableOpacity
        onPress={[Function]}
      >
        <mockConstructor
          color="#545F7A"
          name="trash-can"
          size={40}
        />
      </TouchableOpacity>
    </View>
  </View>
</View>
`;

exports[`LocationCard Tests rendering LocationCard LocationCard with valid input 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffffff",
      "flexDirection": "row",
      "justifyContent": "space-evenly",
      "paddingHorizontal": 10,
      "paddingVertical": 8,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        A1-1
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
  >
    <View>
      <NumericSelector
        isValid={true}
        maxValue={9999}
        minValue={1}
        onDecreaseQty={[Function]}
        onEndEditing={[Function]}
        onIncreaseQty={[Function]}
        onTextChange={[Function]}
        value={12}
      />
    </View>
    <View>
      <TouchableOpacity
        onPress={[Function]}
      >
        <mockConstructor
          color="#545F7A"
          name="trash-can"
          size={40}
        />
      </TouchableOpacity>
    </View>
  </View>
</View>
`;

exports[`LocationCard Tests rendering LocationCard renders the card with locationType reserve and scannerEnabled true 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffffff",
      "flexDirection": "row",
      "justifyContent": "space-evenly",
      "paddingHorizontal": 10,
      "paddingVertical": 8,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#ef5350",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        A1-1
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "color": "#ef5350",
            "fontSize": 12,
            "marginTop": 1,
          }
        }
      >
        [missing "en.LOCATION.PALLET" translation] 1234
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
  >
    <View>
      <NumericSelector
        isValid={true}
        maxValue={9999}
        minValue={0}
        onDecreaseQty={[Function]}
        onEndEditing={[Function]}
        onIncreaseQty={[Function]}
        onTextChange={[Function]}
        value={12}
      />
    </View>
    <View>
      <TouchableOpacity
        onPress={[Function]}
      >
        <mockConstructor
          color="#545F7A"
          name="trash-can"
          size={40}
        />
      </TouchableOpacity>
    </View>
  </View>
</View>
`;

exports[`LocationCard Tests rendering LocationCard renders the card with locationType reserve, scannerEnabled true and scanned true 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffffff",
      "flexDirection": "row",
      "justifyContent": "space-evenly",
      "paddingHorizontal": 10,
      "paddingVertical": 8,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#33691E",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        A1-1
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "color": "#33691E",
            "fontSize": 12,
            "marginTop": 1,
          }
        }
      >
        [missing "en.LOCATION.PALLET" translation] 1234
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
  >
    <View>
      <NumericSelector
        isValid={true}
        maxValue={9999}
        minValue={0}
        onDecreaseQty={[Function]}
        onEndEditing={[Function]}
        onIncreaseQty={[Function]}
        onTextChange={[Function]}
        value={12}
      />
    </View>
    <View>
      <TouchableOpacity
        onPress={[Function]}
      >
        <mockConstructor
          color="#545F7A"
          name="trash-can"
          size={40}
        />
      </TouchableOpacity>
    </View>
  </View>
</View>
`;
